documentation/_posts/2016-06-07-process-api.md

```diff
@@ -489,14 +564,23 @@ The resulting output stream would have lost the sub-stream brackets:
 5) closeBracket, $outtermost
 ```
 
-
 See [noflo-packets/Compact](https://raw.githubusercontent.com/aretecode/noflo-packets/549b67f9e500a958e0283f9d4b8308b43aac66d7/components/Compact.coffee) for an example using [hasDataStream](#has-data-stream) and [getDataStream](#get-data-stream).
 
 -----------------------------------------------------
 # <a id="firing-patterns"></a> Firing Patterns
 
 There are two standard firing patterns
 
+<!-- OK, I'm a bit confused... so there're two patterns:
+
+  1. full stream: < data, data, data, ... >
+  2. per packet: < data, data, data > (ignoring all open/close brackets)
+
+  Is that right? Or do we have another pattern:
+
+  3. per packet with no open/close brackets: data, data, data, ...
+-->
```

Comment:
#3 is the same as #2, because they are firing patterns.

So #2 (if it has brackets or it does not since it ignores all brackets) fires on every data packet.